src/tfwebserver/webserver.cr

```diff
@@ -333,6 +333,33 @@ module TFWeb
       self.handle_update(env, name, true)
     end
 
+    post "/:name/github_webhooks" do |env|
+      name = env.params.url["name"]
+      body = env.params.json
+
+      pushed_branch = body["ref"].to_s.split("/")[-1]
+      default_branch = ""
+
+      # Get the default branch from toml file 
+      if Config.wikis.has_key?(name)
+        default_branch = Config.wikis[name].branch
+      elsif Config.websites.has_key?(name)
+        default_branch = Config.websites[name].branch
+      elsif Config.blogs.has_key?(name)
+        default_branch = Config.blogs[name].branch
+      elsif Config.datasites.has_key?(name)
+        default_branch = Config.datasites[name].branch
+      end
+
+      # Check Signature of the Webhook to insure it is secure
+      signature = "sha1=" + OpenSSL::HMAC.hexdigest(:sha1, ENV["GITHUB_WEBHOOK_SECRET"], body.to_json)
```

Comment:
IMO, Getting the webhook secret from env variables won't work here, because we have multiple Github repositories configured per site, so I think maybe it's better to add it as an optional field in site configuration.

You can find it at `sitetypes.cr`.

Reply:
Unless we use the same secret for all repos!

Reply:
Another small comment, if we are going to use env to pass the secret, we need to check if it's there, and maybe log/raise anything indication it should be configured.

Reply:
This isn't the env variable "http", it is ENV regrading to environment variable used by export

Reply:
Yes, I know and already mentioned that in the first comment, what about multiple repos? have a look at the configuration that's similar too the current configuration file used for deployment:

https://github.com/crystaluniverse/publishingtools/blob/development/deployment/threefoldwebsitesandwikis.toml

I also mentioned in the last comment that if all the repos will use the same webhook secret, then we need to make sure that this environment variable is there, and fail otherwise.